### DISCUSSION TITLE
Ability to delete itself after execution

### DISCUSSION THREAD

**Discussion by jsdjjsd** [Velocidex/velociraptor#2076] [Ideas]
Especially useful when executing over EDR to push collection archives to an SFTP or cloud storage, as soon as the Velociraptor offline collector finishes its acquisition, if self-destruct is enabled via a flag, it ships the archive to the SFTP/cloud storage and then deletes itself.

**Comment by scudette** [comment]
It is hard for an executable to delete itself because the binary is locked during execution :-) It is possible to schedule a delete later or via a script.

Normally you would delete using the same mechanism you copied the file there - or you can place the executable on a network share in which case you dont need to delete it.

See this https://docs.velociraptor.app/docs/deployment/clients/#agentless-deployment for example.
